test/new-e2e/tests/process/testing.go

```diff
@@ -0,0 +1,126 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package process contains end-to-end tests for the general functionality of the process agent.
+package process
+
+import (
+	_ "embed"
+	"encoding/json"
+	"testing"
+
+	agentmodel "github.com/DataDog/agent-payload/v5/process"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/DataDog/datadog-agent/test/fakeintake/aggregator"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e/client"
+)
+
+//go:embed config/process_check.yaml
+var processCheckConfigStr string
+
+//go:embed config/process_discovery_check.yaml
+var processDiscoveryCheckConfigStr string
+
+//go:embed config/system_probe.yaml
+var systemProbeConfigStr string
+
+// assertRunningChecks asserts that the given process agent checks are running on the given VM
+func assertRunningChecks(t *assert.CollectT, vm client.VM, checks []string, withSystemProbe bool) {
+	status := vm.Execute("sudo datadog-agent status --json")
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
As you call this from several tests, suggest using `t.Helper`: it will flag it as an helper testing function, test logger will report it with file name and line of the first `Test` caller
```suggestion
    t.Helper()
	status := vm.Execute("sudo datadog-agent status --json")
```

Reply:
Unfortunately, this function is used in an `EventuallyWithT()` call, which means `t` is actually a `CollectT` which doesn't have the `Helper()` method.